experimental/DoubleAndHyperComplexes/src/Morphisms/ext.jl

```diff
@@ -0,0 +1,415 @@
+struct HomChainFactory{ChainType} <: HyperComplexChainFactory{ChainType}
+  d::AbsHyperComplex
+  c::AbsHyperComplex
+  auto_extend::Bool
+
+  function HomChainFactory(
+      ::Type{ChainType}, d::AbsHyperComplex, c::AbsHyperComplex;
+      auto_extend::Bool=false
+    ) where {ChainType}
+    return new{ChainType}(d, c, auto_extend)
+  end
+end
+
+function (fac::HomChainFactory)(hc::AbsHyperComplex, i::Tuple)
+  I = collect(i)
+  d = fac.d
+  c = fac.c
+  i1 = I[1:dim(d)]
+  i2 = I[dim(d)+1:end]
+  return hom(d[Tuple(-i1)], c[Tuple(i2)])[1]
```

Comment:
Why do you need to `collect(i)` here and elsewhere? Can't you just do
```suggestion
  d = fac.d
  c = fac.c
  i1 = i[1:dim(d)]
  i2 = i[dim(d)+1:end]
  return hom(d[-i1], c[i2])[1]
```

Reply:
I must have thought that the indexing with a range doesn't work for `Tuple`s. But apparently I was wrong.